src/util/unicode.cpp

```diff
@@ -20,6 +19,24 @@ Author: Daniel Kroening, kroening@kroening.com
 
 /*******************************************************************\
 
+Function: is_little_endian_arch
+
+  Inputs:
+
+ Outputs: True if the architecture is little_endian
+
+ Purpose: Determine endianness of the architecture
+
+\*******************************************************************/
+
+bool is_little_endian_arch()
+{
+  uint16_t i=1;
```

Comment:
I think this should not be target-analysis architecture, but the architecture on which cbmc runs, because the underlying architecture determines how integers are endianed.

I don't follow the comment on bytes/chars. uint8_t should be 8 bits, or not?